### PR TITLE
Stop reporting ReplicaSets

### LINKED ISSUE #2662: don't report on replicasets from probe
In #2661 we removed replicasets from the display, instead adding deployments as direct parents of pods. Once we are sure that is the right way to go we should in fact remove replicasets from the probes, performing the mapping to deployments there, and convert old reports to the new format in `report.Upgrade()` for backward compatibility.

That should improve performance in several ways:
- reduced report size, and hence probe report writing, app report reading, and app report storage performance - replicasets constitute a monotonically growing data set, since during normal operation the replicasets of old deployment generations are never removed
- improved rendering performance since we no longer need to do the mapping there

### PR DESCRIPTION
**Pull request by rbruggem**
Also, add Deployment as Pod parent.
Fixes #2662

### CODE REVIEW

File: probe/kubernetes/reporter.go [hunk 1]
```diff
@@ -446,7 +393,7 @@ func match(namespace string, selector labels.Selector, topology, id string) func
 	}
 }
 
-func (r *Reporter) podTopology(services []Service, replicaSets []ReplicaSet, daemonSets []DaemonSet, statefulSets []StatefulSet, cronJobs []CronJob) (report.Topology, error) {
+func (r *Reporter) podTopology(services []Service, daemonSets []DaemonSet, statefulSets []StatefulSet, cronJobs []CronJob, deployments []Deployment) (report.Topology, error) {
```
Comment (rade) [inline probe/kubernetes/reporter.go]: pls put `deployments` in the same place where `replicaSets` used to be.

File: probe/kubernetes/reporter.go [hunk 2]
```diff
@@ -474,16 +421,16 @@ func (r *Reporter) podTopology(services []Service, replicaSets []ReplicaSet, dae
...
 			return pods, err
 		}
 		selectors = append(selectors, match(
-			replicaSet.Namespace(),
+			deployment.Namespace(),
 			selector,
-			report.ReplicaSet,
-			report.MakeReplicaSetNodeID(replicaSet.UID()),
+			report.Deployment,
+			report.MakeDeploymentNodeID(deployment.UID()),
```
Comment (rade) [inline probe/kubernetes/reporter.go]: How does this work?

Comment (rbruggem) [reply probe/kubernetes/reporter.go]: `match` will add the deployment as the pod's parent if the following conditions are true:
* the pod and the deployment belong to the same namespace
* the pod and the deployment match the same labels.

Comment (rbruggem) [reply probe/kubernetes/reporter.go]: https://github.com/weaveworks/scope/blob/master/probe/kubernetes/reporter.go#L403

Comment (rbruggem) [reply probe/kubernetes/reporter.go]: Deployments, ReplicaSets and Pods will be assigned the same label in the manifest.
The label is required: https://github.com/kubernetes/kubernetes/issues/28475

**Pull request merged**